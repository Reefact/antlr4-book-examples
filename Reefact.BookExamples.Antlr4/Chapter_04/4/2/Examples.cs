#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using NFluent;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_04._4._2 {

    [UseReporter(typeof(CustomReporter))]
    public class Examples {

        [Fact]
        public void mermaid_style_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("2 9 10 3 1 2 3");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void lisp_tree_style() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("2 9 10 3 1 2 3");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Check.That(lispStyleTree).IsEqualTo("(file (group 2 (sequence 9 10)) (group 3 (sequence 1 2 3)))");
        }

    }

}