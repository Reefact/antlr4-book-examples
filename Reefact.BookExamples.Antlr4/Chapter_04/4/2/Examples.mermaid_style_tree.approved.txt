graph TD
	1["file"] --> 2["group"]
	2 --> 3["2"]
	2 --> 4["sequence"]
	4 --> 5["9"]
	4 --> 6["10"]
	1 --> 7["group"]
	7 --> 8["3"]
	7 --> 9["sequence"]
	9 --> 10["1"]
	9 --> 11["2"]
	9 --> 12["3"]
